### ISSUE TITLE
Feature: Auto-Create Child RW-Parking for OSE

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#3640]
### As an OSE reviewer, I'd like the system to auto-create a child parking permit folder when the folder info 'Parking Reservations' value is 'Yes'.

**Stakeholder:** Maria A
**DTS Manager:** Nate

**Acceptance Criteria:**
- This feature applies to RW-Street Event folder
- When the value of folder info 75087 'Parking Reservations' is updated to Yes, the system checks to to see if a child RW-Parking folder exists.
- If a child RW-Parking folder does not exist and the permit Analyst Accepts the "Event review" Process, the system will create the folder automatically, and create the parent/child relationship. 
- The system should copy the main property from the parent folder (not all properties need to be copied over)
- The system should copy the following from the parent to the child Parking folder: 
             - Folder Name
             - Folder Description 
             - Folder Properties
             - People Record of type 'Applicant' 
             - Folder Info "Number of days Requested"
             - Folder Info "Event Start Date"
             - Folder Info "Event End Date"
- The newly inserted RW-Parking folder should have a Folder Status of 'Pending'

**Comment by TracyLinder** [comment]
**Next Step:** Perform scoping w/ stakeholder, finalize acceptance criteria.

**Comment by vschandramouli** [comment]
Edited/Updated AC based off meeting with Maria. Ready for review by @Nadin-Nader

**Comment by Nadin-Nader** [comment]
@vschandramouli @TracyLinder Acceptance criteria looks good, easy to understand and straight forward. Moving to Backlog

**Comment by aishwarya2505** [comment]
I am looking in to this one. Will begin code work shortly.

**Comment by aishwarya2505** [comment]
Coding is complete and one of the test scenarios failed. I am working on resolving the bug.

**Comment by aishwarya2505** [comment]
4/28 - E-mailed the DEV Deployment Plan on 4/14/2021 - Commit ID: 27d70bd4502f

**Comment by aishwarya2505** [comment]
4/28 - 7c915a5eb313 Commit made. Will send the DEV Deployment Plan post CTM Approval

**Comment by Nadin-Nader** [comment]
@aishwarya2505 @Sreelatha-BlazeKannan is the Deployment plan ready for this?

## **Status:**

Waiting on DEV Deployment plan to send to DevOps CTM

**Comment by aishwarya2505** [comment]
@Nadin-Nader DEV deployment is ready and emailed. Thank you.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @aishwarya2505  I spoke to Maria and asked her the questions you needed answers to, See below:

> While testing #3640 I see that it creates a child folder when parking reservations folder info is “y” for RW-Street event folder. However I would want know whether while creating the folder in automatic add phase also do we need to create a child folder if the Parking reservations folder info is “y” ? And if there are more than one property in parent folder then do we need to copy all properties to child folder also? 

> While testing #3640 I see that it creates a child folder when parking reservations folder info is “y” for RW-Street event folder. However I would want know whether while creating the folder in automatic add phase also do we need to create a child folder if the Parking reservations folder info is “y” ?

_**Maria said that the child folder will need to be created when the Permit Analyst accepts the the "Event Intake" Process and not during Web Acceptance process. In summary, The info NEEDS to be  "YES" AND the "Event Intake" process needs to be accepted by the Permit Analyst in order for the creation of the child folder.**_

` And if there are more than one property in parent folder then do we need to copy all properties to child folder also? `

**_As for all of the properties, She is ok with it just having the Main property from the parent folder,  instead of all._**

I updated the Acceptance criteria by adding the ne parameters set by Maria.

@Sreelatha-BlazeKannan @aishwarya2505 let me know if this makes sense or not.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan This has been deployed to DEV

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan UATd this with Maria and it looks good to go to TEST.

## **Status:**

## **Approved for TEST Deployment🎉🎉🙌**

**Comment by sreelathab** [comment]
@Nadin-Nader , @TracyLinder  - Sent TEST Deployment Plan on 05/14/2021. Here is the commit ID: 0906e02520ed and Service Now Request #SCTASK0137673

**Comment by Nadin-Nader** [comment]
@Nadin-Nader UAT with Kim and Maria

**Comment by sreelathab** [comment]
@Nadin-Nader , @TracyLinder  - Tested in TEST Environment Working as expected. and Ready for TEST UAT.

**Comment by Nadin-Nader** [comment]
Scheduled UAT on TEST for Monday 5/24/21 @930am

**Comment by Nadin-Nader** [comment]
Spoke to Maria and we agreed to open a new card for the Electrical fee to be taken out. We are leaving the ATD Surcharge fee.

**Comment by Nadin-Nader** [comment]
**Status:** 
This has been sent to the DevOps CM Team to deploy to PROD under task # **SCTASK0140802**

**Comment by Nadin-Nader** [comment]
## **Status:**

Still waiting on PROD Deployment

**Comment by Nadin-Nader** [comment]
## **Status:** 
This has been promoted to PROD per Martin. I will send an email to the Stakeholders to let them know.

**Comment by Nadin-Nader** [comment]
Email to Maria:

> Hi Maria,
> 
> Hope you are doing well. Hey I wanted to let you know that the card for the Feature: Auto-Create Child RW-Parking for OSE; #3640, is on PROD (has been on PROD since Friday).
> I will keep it open for a few days so that you and your team can UAT it properly on PROD. Please let me know if there are any issues. I will follow up on Wednesday to see how things are going.
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> IT Business Systems Analyst
> Data and Technology Services
> ————————————————————————————
> transportation.data@austintexas.gov 🚦 austinmobility.io

**Comment by Nadin-Nader** [comment]
Got a response from Ryan asking if this was the report we went over the other day in the meeting. I responded with yes and to let me know tomorrow if the report is good to go on PROD. If it is i will close the card. Email below:

> Yes sir it is. Let me know tomorrow if everything is good and I will close the card
> 
> From: Mooney, Ryan <Ryan.Mooney@austintexas.gov> 
> Sent: Wednesday, June 9, 2021 2:38 PM
> To: Nader, Nadin <Nadin.Nader@austintexas.gov>; Ujano, Darren <Darren.Ujano@austintexas.gov>
> Cc: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov>; Linder, Tracy <Tracy.Linder@austintexas.gov>
> Subject: RE: Permitting Metrics INSPECTIONS Report 4 #5579
> 
> This is the one we went over earlier this week right?

**Comment by Nadin-Nader** [comment]
This deployment is working as expected. Closing
